python/TestHarness/TestHarness.py

```diff
@@ -464,143 +524,143 @@ def checkPBSEmulator(self):
+                    # to go muddling around changing specs like this.
+                    tester.specs['test_dir'] = self.pbs_data[job_id.group(1)]['test_dir']
+                    output = tester.processResults(tester.specs['moose_dir'], exit_code, self.options, outfile)
+                    self.testOutputAndFinish(tester, exit_code, outfile)
+                    return
+                else:
+                    # I ran into this scenario when the cluster went down, but launched/completed my job :)
+                    reason = 'FAILED (NO STDOUT FILE)'
+                    tester.setStatus(reason, tester.bucket_fail)
+            elif output_value.group(1) == 'R':
+                # Job is currently running
+                reason = 'RUNNING'
+            elif output_value.group(1) == 'E':
+                # Job is exiting
+                reason = 'EXITING'
+            elif output_value.group(1) == 'Q':
+                # Job is currently queued
+                reason = '%s QUEUED' % (job_id.group(1))
+            elif output_value.group(1) == 'H':
+                # Job is currently holding for another job
+                # The else is there, because I ran into an interesting race condition where
+                # PBS reported HOLD... but during a state it didn't know why (and when I checked
+                # again, this test was now running). Transitional state? whatever GO PBS!
+                if holding:
+                    reason = '%s HOLDING for %s' % (job_id.group(1), holding.group(1))
+                else:
+                    reason = 'HOLD'
         else:
-            print 'PBS batch file not found:', self.options.pbs_cleanup
-            sys.exit(1)
+            # Job status not available
+            reason = 'INVALID QSTAT RESULTS'
+            tester.setStatus(reason, tester.bucket_fail)
+            self.handleTestResult(tester, output, reason, start=0, end=0, add_to_table=True)
 
-        # Loop through launched jobs and delete whats found.
-        for job in batch_list:
-            if os.path.exists(job[2]):
-                batch_dir = os.path.abspath(os.path.join(job[2], os.pardir)).split('/')
-                if os.path.exists('/'.join(batch_dir)):
-                    shutil.rmtree('/'.join(batch_dir))
-                if os.path.exists('/'.join(batch_dir[:-1]) + '/' + self.options.pbs_cleanup + '.cluster'):
-                    os.remove('/'.join(batch_dir[:-1]) + '/' + self.options.pbs_cleanup + '.cluster')
-        os.remove(self.options.pbs_cleanup)
-    # END PBS Defs
+        if reason != '' and tester.getStatus() != tester.bucket_fail:
+            tester.setStatus(reason, tester.bucket_pbs)
+            self.handleTestStatus(tester)
 
     ## Method to print output generated by the TestHarness while attempting to run a tester
     def handleTestStatus(self, tester, output=None):
         status = tester.getStatus()
-        test_completed = False
+        test_completed = 0
```

Comment:
I am playing off the fact a zero is seen as a False, while any positive int is a True. Which will always be either a PBS Job ID, or well... a failed launch.